source/Account.cpp

```diff
@@ -257,7 +258,10 @@ string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
 
 	// If you didn't make any payments, no need to continue further.
 	if(!(salariesPaid + maintenancePaid + mortgagesPaid + finesPaid + debtPaid))
+	{
+		totalPreviousPayment = 0;
```

Comment:
Why not reset the value in one place instead of two? There aren't any other early returns, so the value is reset every time this function is run. Doing it here like this and then again in the other path rather than at the beginning of the function, for example, doesn't make it clear that it is always reset until you establish that there are no earlier returns. I think resetting it at the beginning makes how it is used much clearer.

Since we don't actually need its value to persist across steps (it is only ever retrieved moments after running this function), I'd actually rather it be a local and included in the return value.